assets/js/script.js

```diff
@@ -0,0 +1,107 @@
+import { UI } from "./ui.js";
+const http = new Http();
+
+const fetchDataFromApi = {
+  getData() {
+    return http.get("products").then((data) => data);
+  },
+};
+
+// Fetch and populate product data
+function fetchProductDataHandler() {
+  fetchDataFromApi.getData().then((data) => {
+    UI.populateCarousel(data);
+    UI.populateProducts(data);
+  });
+}
+
+// Variable for selected item ID to delete
+let itemIdToDelete = "";
+const onDelete = (event) => deleteProductHandle(event, itemIdToDelete);
+
+UI.btnAddProduct.addEventListener("click", addNewProductHandler);
+UI.btnSubmitProduct.addEventListener("click", submitProductHandler);
+UI.productList.addEventListener("click", editOrDeleteProductModalHandler);
+UI.modalContainer.addEventListener("click", closeModalHandler);
+
+function addNewProductHandler() {
+  UI.showModal(true, "form");
+  UI.setButtonState("add");
+}
+
+function closeModalHandler(e) {
+  if (e.target.classList.contains("js-cancel")) {
+    UI.showModal(false, "form");
+    UI.clearFields();
+    UI.btnDeleteProduct.removeEventListener("click", onDelete);
```

Comment:
Do you need to unregister and re-add the event listener if you have only one button to trigger delete? 🤔  
Events usually need unregistering when the component which was added is being removed.

Reply:
In other case I have that bug what we was talking about. On delete previous canceled elements also removes. :/

Reply:
Ok, could be the case, I possibly haven't tracked down all the logic in my head. 😉 👍